downstairs/src/dump.rs

```diff
@@ -0,0 +1,267 @@
+             * If we are looking at one extent in detail, we skip all the
+             * others.
+             */
+            if let Some(ce) = cmp_extent {
+                if en != ce {
+                    continue;
+                }
+            }
+            let inner = e.inner();
+
+            /*
+             * Create the ExtentMeta struct for this directory's extent
+             * number
+             */
+            let extent_info = ExtentMeta {
+                ext_version: 0,
+                gen_number: inner.gen_number().unwrap(),
+                flush_number: inner.flush_number().unwrap(),
+                dirty: inner.dirty().unwrap(),
+            };
+
+            /*
+             * If we have an entry already, then add this at our directory
+             * index.  If we don't the create the hashmap for this index,
+             * then add the extent_info to it.
+             */
+            let ei = all_extents.entry(en).or_default();
+            ei.ei_hm.insert(index as u32, extent_info);
+        }
+    }
+
+    /*
+     * If we just want one extent, then go and handle that now.
+     * TODO: Support a list of extents to display.
+     * TODO: Support a specific compare option, as maybe we don't want to
+     *       compare every time.
+     */
+    if let Some(ce) = cmp_extent {
+        if ce >= total_extents {
+            bail!(
+                "Requested extent {} is a higher index than valid ({})",
+                ce,
+                total_extents,
+            );
+        }
+        if dir_count < 2 {
+            bail!("Need more than one region directory to compare data");
+        }
+        let en = all_extents.get(&ce).unwrap();
+        show_extent(region_dir, &en.ei_hm, ce, blocks_per_extent, block_size)?;
+
+        return Ok(());
+    };
+
+    /*
+     * Print out the extent info one extent at a time, in order
+     */
+    let mut ext_num = all_extents.keys().collect::<Vec<&u32>>();
+    ext_num.sort_unstable();
+
+    print!("N");
+    for _ in 0..dir_count {
+        print!("      GEN FLUSH_ID D");
+    }
+    println!();
+
+    for en in ext_num.iter() {
+        print!("{} ", en);
+        if let Some(ei) = all_extents.get(en) {
+            for dir_index in 0..dir_count {
+                if let Some(em) = ei.ei_hm.get(&(dir_index as u32)) {
+                    let dirty;
+                    if em.dirty {
+                        dirty = "D".to_string();
+                    } else {
+                        dirty = " ".to_string();
+                    }
+                    print!(
+                        "{:8} {:8} {} ",
+                        em.gen_number, em.flush_number, dirty
+                    );
+                } else {
+                    print!("-");
+                }
+            }
+        } else {
+            println!("No data for {}", en);
+        }
+        println!();
+    }
+
+    Ok(())
+}
+
+/*
+ * Show the metata and a block by block diff of a single extent
```

Comment:
metadata

Reply:
Fixed in 4dd9a19ac0e101bdd9a8c14f01b7deaef775d55f